react/features/base/user-interaction/middleware.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import { APP_WILL_MOUNT, APP_WILL_UNMOUNT } from '../app';
+import { MiddlewareRegistry } from '../redux';
+
+import {
+    SET_USER_INTERACTION_LISTENER,
+    USER_INTERACTION_RECEIVED
+} from './actionTypes';
+
+/**
+ * Implements the entry point of the middleware of the feature base/user-interaction.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case APP_WILL_MOUNT:
+        _startListeningForUserInteraction(store);
+        break;
+
+    case APP_WILL_UNMOUNT:
+    case USER_INTERACTION_RECEIVED:
+        _stopListeningForUserInteraction(store);
+        break;
+    }
+
+    return next(action);
+});
+
+/**
+ * Registers listeners to notify redux of any user interaction with the page.
+ *
+ * @param {Object} store - The redux store.
+ * @private
+ * @returns {void}
+ */
+function _startListeningForUserInteraction(store) {
+    const userInteractionListener = event => {
```

Comment:
I usually try to avoid defining nested functions. Can't we avoid it somehow? Maybe define it as a function and use bind for the store? WDYT?

Note: If you prefer the nested function approach please ignore the comment.

Reply:
We use this pattern extensively on mobile when registering event handlers in the store, FWIW.

Reply:
I can't figure out a way to remove the handler after the first event without using this approach.